### PR TITLE
[ws-proxy] Reload to restart

### PR DESCRIPTION
**Pull request by csweichel**
This PR brings the "reload to restart" feature back to ws-proxy, cleans up the central `routes.go, makes unit tests for the routes easier and increases ws-proxy test coverage from 50% to > 60%.

### How to test
1. Start a workspace
2. Stop the workspace using the command palette
3. Once the workspace is stopped, reload the tab. The workspace should start again.

### CODE REVIEW

**Comment by akosyakov** [comment]
It fixes https://github.com/gitpod-io/gitpod/issues/2085

File: components/ws-proxy/pkg/proxy/routes.go [hunk 1]
```diff
@@ -67,113 +67,84 @@ func installTheiaRoutes(r *mux.Router, config *RouteHandlerConfig, ip WorkspaceI
...
+		return
 	}
-	// permament redirect to tell the browser to cache redirect request and don't ask the server again
-	http.Redirect(w, req, redirectURL, 308)
-}
 
-// SupervisorIDEHostHandler serves supervisor's IDE host
-func SupervisorIDEHostHandler(r *mux.Router, config *RouteHandlerConfig) {
+	r := route.Subrouter()
 	r.Use(logRouteHandlerHandler("SupervisorIDEHostHandler"))
```
Comment (akosyakov) [inline components/ws-proxy/pkg/proxy/routes.go:148]: `HandleSupervisorFrontendRoute`

File: components/ws-proxy/pkg/proxy/routes.go [hunk 2]
```diff
@@ -393,22 +356,19 @@ func staticTheiaResolver(config *Config, req *http.Request) (url *url.URL, err e
...
 
-func dynamicTheiaResolver(infoProvider WorkspaceInfoProvider) targetResolver {
-	return func(config *Config, req *http.Request) (res *url.URL, err error) {
-		coords := getWorkspaceCoords(req)
-		info := infoProvider.WorkspaceInfo(coords.ID)
-		if info == nil {
-			log.WithFields(log.OWI("", coords.ID, "")).Warn("no workspace info available - cannot resolve Theia route")
-			return nil, xerrors.Errorf("no workspace information available - cannot resolve Theia route")
-		}
+func dynamicTheiaResolver(config *Config, req *http.Request) (res *url.URL, err error) {
```
Comment (akosyakov) [inline components/ws-proxy/pkg/proxy/routes.go:359]: `theia` still pops up here and there in function names and comments which are relevant to IDEs in general, not sure whether it was an intention of the PR to clean it up too

**Review by akosyakov — APPROVED** [review]
It works as advertised and code changes looks good. Thank you for adding more tests!

File: components/ws-proxy/pkg/proxy/routes.go (hunk 2 above)
Comment (csweichel) [reply components/ws-proxy/pkg/proxy/routes.go:359]: it wasn't, but maybe it should :)

Comment (csweichel) [reply components/ws-proxy/pkg/proxy/routes.go]: it still isn't all gone, but it's better now

**Pull request merged**